### ISSUE TITLE
CAn't upgrade the tool to 1.5

### ISSUE THREAD

**Issue by alpha028** [FezVrasta/ark-server-tools#556]
Hi, I can't upgrade the tools from 1.4 to 1.5, it gives me the error : 
#arkmanager upgrade-tools
arkmanager v1.4: Checking for updates...
A new version was found! Do you want to upgrade ARK Server Tools to v1.5?Y
Unable to retrieve latest commit

I use it as sudo user.

I have to update it because everytime I want to update the mods, it want to download only the TC primitive+ as mods, so it fails then exit the update

**Comment by klightspeed** [comment]
What does the following command print:
```
curl -L -k https://api.github.com/repos/FezVrasta/ark-server-tools/git/refs/heads/master
```

**Comment by boistordu** [comment]
I'm running the docker version of ark of turzam with ark manager installed within.  I ve still the download mod 11111 problem so I ve tried to do the arkmanager upgrade-tools but it loops on checking for update.  
I ve done your curl command and the results :
 "ref": "refs/heads/master",                                                       
  "url": "https://api.github.com/repos/FezVrasta/ark-server-tools/git/refs/heads/mas
ter",                                                                               
  "object": {                                                                       
    "sha": "8a59a8c78c2b1cc052207474e5be132a54fb10b5",                              
    "type": "commit",                                                               
    "url": "https://api.github.com/repos/FezVrasta/ark-server-tools/git/commits/8a59
a8c78c2b1cc052207474e5be132a54fb10b5"                                               
  }

**Comment by boistordu** [comment]
I was already on 1.5 by the way so I don't know if there are several arkmanager version...  I don't know either if the upgrade-tools work for just some commit?

**Comment by klightspeed** [comment]
@alpha028 have you fixed your issue?  The "Unable to retrieve latest commit" suggests that the `curl` command wasn't working.

@boistordu what is the output of the `arkmanager upgrade-tools` command?

@alpha028 @boistordu does the install command (`curl -sL http://git.io/vtf5N | sudo bash -s steam`) work?

**Comment by alpha028** [comment]
Still havent fix it. But havent try a lot since i deleted the folder 111111 in the mod directory of ark, the update mpd command is working perfectly.
I am busy at work so ill tell you when ill have time to test it again

**Comment by boistordu** [comment]
Here is the output of you command.

T?l?charger Outlook pour Android<https://aka.ms/ghei36>

De : alpha028
Envoy? : mardi 16 ao?t 23:36
Objet : Re: [FezVrasta/ark-server-tools] CAn't upgrade the tool to 1.5 (#556)
? : FezVrasta/ark-server-tools
Cc : boistordu, Mention

Still havent fix it. But havent try a lot since i deleted the folder 111111 in the mod directory of ark, the update mpd command is working perfectly.
I am busy at work so ill tell you when ill have time to test it again

-

**Comment by klightspeed** [comment]
@boistordu if you attached an image, it did not come through.

**Comment by boistordu** [comment]
Let me 35 minutes and I will run every test you want.  I was sending an image from ssh terminal from my phone.
So anyway you didn t answer me:
Are there different version of ark manager?  Different maintainer I mean
And also does the arkmanager upgrade for small commit?

Télécharger Outlook pour Android<https://aka.ms/ghei36>



On Tue, Aug 16, 2016 at 11:57 PM +0200, "Ben Peddell" <notifications@github.com<mailto:notifications@github.com>> wrote:


@boistordu<https://github.com/boistordu> if you attached an image, it did not come through.

-

**Comment by klightspeed** [comment]
This repository (https://github.com/FezVrasta/ark-server-tools) is the official home of arkmanager.

`arkmanager upgrade-tools` should work for both version upgrades and for minor fixes (hotfixes).

**Comment by boistordu** [comment]
Okey that answered to my questions :) So I can in confidence post my problem here.
So theorically, this docker is made without steamcmd. I didn't investigate yet how it has install the arkserver so be gentle with me. 
the only output i've got from your upgrade-tools command is : `arkmanager v1.5: Checking for updates...`
So I don't know if you plan some logs about it? And where are they ?
the output of your first test is  : 
`root@d9e593cec28f:/ark# curl -L -k https://api.github.com/repos/FezVrasta/ark-server-tools/git/refs/heads/master 
{
  "ref": "refs/heads/master",
  "url": "https://api.github.com/repos/FezVrasta/ark-server-tools/git/refs/heads/master",
  "object": {
    "sha": "8a59a8c78c2b1cc052207474e5be132a54fb10b5",
    "type": "commit",
    "url": "https://api.github.com/repos/FezVrasta/ark-server-tools/git/commits/8a59a8c78c2b1cc052207474e5be132a54fb10b5"
  }
}`

the output of your second command is : 
`root@d9e593cec28f:/ark# curl -sL http://git.io/vtf5N | sudo bash -s steam
WARNING: A previous version of ARK Server Tools was detected in your system, your old configuration was not overwritten. You may need to manually update it.
ARK Server Tools were correctly installed in your system inside the home directory of steam!`

**Comment by klightspeed** [comment]
> the only output i've got from your upgrade-tools command is : `arkmanager v1.5: Checking for updates...`

The fact that the other curl commands you've run do work suggests that the following command is hanging while trying to connect:
```
curl https://raw.githubusercontent.com/FezVrasta/ark-server-tools/master/.version
```

> the output of your second command is : 
```
root@d9e593cec28f:/ark# curl -sL http://git.io/vtf5N | sudo bash -s steam
WARNING: A previous version of ARK Server Tools was detected in your system, your old configuration was not overwritten. You may need to manually update it.
ARK Server Tools were correctly installed in your system inside the home directory of steam!
```

That should have successfully updated arkmanager to the lastest stable version.

**Comment by boistordu** [comment]
I guess that you didn't put any logs about the upgrade to see a hypothetic problem of timeout or something like that? Or maybe is that happening so often that's okey?

**Comment by klightspeed** [comment]
Please run `arkmanager --version` and check that the commit shown is 8a59a8c

**Comment by boistordu** [comment]
yeah i've looked a few horus ago and it was exactly that. Thanks anyway, any ideas why the connection hangs like that?

**Comment by boistordu** [comment]
Hi, 

I have again the same problem:
```
root@28af7e1a3035:/ark# arkmanager upgrade-tools  
arkmanager v1.6: Checking for updates...
A new version was found! Do you want to upgrade ARK Server Tools to v404: Not Found?n
root@28af7e1a3035:/ark# curl https://raw.githubusercontent.com/FezVrasta/ark-server-tools/master/.version
1.6
root@28af7e1a3035:/ark# arkmanager --version
Version: 1.6
Channel: 
Blob SHA: f6400eee6bcb444dabe56a6f07890b34dce7e6cf
root@28af7e1a3035:/ark# curl -L -k https://api.github.com/repos/FezVrasta/ark-server-tools/git/refs/heads/master
{
  "ref": "refs/heads/master",
  "url": "https://api.github.com/repos/FezVrasta/ark-server-tools/git/refs/heads/master",
  "object": {
    "sha": "95fe90b15f5dd9c98aebc4f94a1f8fb399776117",
    "type": "commit",
    "url": "https://api.github.com/repos/FezVrasta/ark-server-tools/git/commits/95fe90b15f5dd9c98aebc4f94a1f8fb399776117"
  }
}
```

**Comment by boistordu** [comment]
```
root@f67ce8583f09:/home/steam/ark-server-tools# arkmanager upgrade-tools
arkmanager v1.6: Checking for updates...
A new version was found! Do you want to upgrade ARK Server Tools to v404: Not Found?y
Downloading installer

gzip: stdin: not in gzip format
tar: Child died with signal 13
tar: Error is not recoverable: exiting now
/usr/local/bin/arkmanager: line 48: cd: ark-server-tools-/tools: No such file or directory
install.sh not found in /ark-server-tools-nW59s54p
```

**Comment by boistordu** [comment]
i've controlled the content about the last commit you made and I have the change. Did you change something about the print of the version ark your tool? Because I don't have the same result SHA than the commit anymore...

**Comment by klightspeed** [comment]
It sounds like you have the `arkstChannel` setting set to either an empty string or a non-existent branch.  Please try setting it to `master`

**Comment by boistordu** [comment]
already done and when I do it tells me that I'm already on master and change it because I use the option -f

**Comment by boistordu** [comment]
and it doesn't change anything by the way

**Comment by boistordu** [comment]
Is it a file where I can put it?

**Comment by klightspeed** [comment]
Open your `/etc/arkmanager/arkmanager.cfg` with a text editor, and check what `arkstChannel` is set to.

**Comment by boistordu** [comment]
empty string. Can I put master in it?

**Comment by klightspeed** [comment]
Please do.

**Comment by boistordu** [comment]
still nothing in channel when I ask for version, should I do something else?

**Comment by klightspeed** [comment]
Check if `~/.arkmanager.cfg` exists, and remove or modify it if necessary.

**Comment by boistordu** [comment]
I've seen your last commit so I recompile my whole thing. 
now if I do this I have : 
```
root@d0bc82e51ea4:/home/steam/ark-server-tools/tools# arkmanager --version
Version: 1.6
Channel: master
Blob SHA: 70e06f8805dd7b1071d53e5f59ae78967cccaab2
```
```

root@d0bc82e51ea4:/home/steam/ark-server-tools/tools# arkmanager upgrade-tools
arkmanager v1.6: Checking for updates...
Channel '${BRANCH}' does not exist

Available channels:
api
fix-323
master
1.2-dev
1.3-dev
1.4-dev
1.5-dev
1.6-dev-rc
1.6-dev.unstable
1.6-dev

root@d0bc82e51ea4:/home/steam/ark-server-tools/tools# 
```

```
root@d0bc82e51ea4:/home/steam/ark-server-tools/tools# curl -L -k https://api.github.com/repos/FezVrasta/ark-server-tools/git/refs/heads/master 
{
  "ref": "refs/heads/master",
  "url": "https://api.github.com/repos/FezVrasta/ark-server-tools/git/refs/heads/master",
  "object": {
    "sha": "f4a3cfd97d3321d99262db03d4c070404d0c2694",
    "type": "commit",
    "url": "https://api.github.com/repos/FezVrasta/ark-server-tools/git/commits/f4a3cfd97d3321d99262db03d4c070404d0c2694"
  }
}
```

**Comment by boistordu** [comment]
I don't see any file with that with find

**Comment by klightspeed** [comment]
That suggests that you have `arkstChannel="${BRANCH}"` in your `arkmanager.cfg`.

**Comment by boistordu** [comment]
yeah guess so, but now I need to find which one it is of arkmanager.cfg

**Comment by boistordu** [comment]
find the wrong file. is it your installation that does a /etc/arkmanager/arkmanager.cfg or my config?

**Comment by boistordu** [comment]
now I have: 

```
root@d0bc82e51ea4:/ark# arkmanager --version    
Version: 1.6
Channel: master
Commit: 2f1d3ca
Release Tag: v1.6.02
Blob SHA: f257805c69547f7f246fcd24cd3be2eedc4cb1bd
```

I don't really understand how git works for the commit when you have branch + gittag if you have a reference that explains it, I would be happy to have it

**Comment by boistordu** [comment]
I would like to know something. My configuration was based on some environment variable... So the branch line was arkstChannel=${BRANCH} 
does it mean then that arkmanager can't work witrh environment variables, since it doesn't read them?

**Comment by klightspeed** [comment]
It's only the variables that are referenced by the upgrade.

If you're installing from a git repo, then we can create a git-install script that installs from the current repo.

**Comment by boistordu** [comment]
no actualli, as I said in one of my other post I think, I'm using a docker container to do that .
It seems that if I use the environmnent variables which I put in the creation of the container , your script doesn't take it in mind and so that why if I put arkstChannel=${BRANCH}  it just read ${BRANCH} and not master from the environment variables....

**Comment by boistordu** [comment]
I will open a new ticker it's better maybe.

**Comment by TheInternetIsReal** [comment]
Just FYI for anyone else looking at this - the issue is within the ark docker itself. The author disabled being able to update arkmanager.

**Comment by otherippo** [comment]
> Just FYI for anyone else looking at this - the issue is within the ark docker itself. The author disabled being able to update arkmanager.

Same here, but I found the docker image from cwollinger which is a fork, and it's fully working and up to date! You can find it here: https://hub.docker.com/r/cwollinger/ark-docker

The used version of Ark Manager depends on the ENV GIT_TAG (e.g. ENV GIT_TAG v1.6.48) in the dockerfile, that's what he replaced (the original value was v.1.5)
